### PR TITLE
mesa: fix build on darwin

### LINKED ISSUE #199220: mesa fails to build on aarch64-darwin
### Steps To Reproduce
1. build *mesa*

### Build log
```
[1077/1078] Linking static target src/glx/libglx.a
[1078/1078] Linking target src/glx/libGL.1.dylib
buildPhase completed in 33 seconds
installing
install flags: install
[1/9] Generating src/git_sha1.h with a custom command
[1/2] Installing files.
Installing src/mapi/shared-glapi/libglapi.0.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib
Installing src/mapi/es1api/libGLESv1_CM.1.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib
Installing src/mapi/es2api/libGLESv2.2.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib
Installing src/gallium/targets/dri/libgallium_dri.so to /nix/store/7pmab9bri6pc4y5phkmfdqlwavmc01jx-mesa-22.2.2-drivers/lib/dri
Installing src/glx/libGL.1.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/KHR/khrplatform.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/KHR/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES/egl.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES/gl.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES/glext.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES/glplatform.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES2/gl2.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES2/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES2/gl2ext.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES2/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES2/gl2platform.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES2/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES3/gl3.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES3/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES3/gl31.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES3/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES3/gl32.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES3/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES3/gl3ext.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES3/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GLES3/gl3platform.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GLES3/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GL/gl.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GL/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GL/glcorearb.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GL/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GL/glext.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GL/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GL/glx.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GL/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GL/glxext.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GL/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/include/GL/internal/dri_interface.h to /nix/store/1ds2sh20vns2zq2zljcgv6kycib4i0nh-mesa-22.2.2-dev/include/GL/internal/
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/src/util/00-mesa-defaults.conf to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/share/drirc.d
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/build/meson-private/glesv1_cm.pc to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/pkgconfig
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/build/meson-private/glesv2.pc to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/pkgconfig
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/build/meson-private/dri.pc to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/pkgconfig
Installing /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/build/meson-private/gl.pc to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/pkgconfig
Installing symlink pointing to libglapi.0.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/libglapi.dylib
Installing symlink pointing to libGLESv1_CM.1.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/libGLESv1_CM.dylib
Installing symlink pointing to libGLESv2.2.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/libGLESv2.dylib
Installing symlink pointing to libGL.1.dylib to /nix/store/xf9hi5h9ynjjpnrmdbijnhg153zr3fnh-mesa-22.2.2/lib/libGL.dylib
Running custom install script '/nix/store/14sv2h3nag638062j87lg3ic4z8ipgis-python3-3.10.8/bin/python3.10 /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/bin/install_megadrivers.py /private/tmp/nix-build-mesa-22.2.2.drv-0/mesa-22.2.2/build/src/gallium/targets/dri/libgallium_dri.so /nix/store/7pmab9bri6pc4y5phkmfdqlwavmc01jx-mesa-22.2.2-drivers/lib/dri swrast_dri.so'
mv: missing file operand
Try 'mv --help' for more information.
```

### Additional context

#196310

### Notify maintainers

@primeos @vcunat

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.1.0, macOS 13.0`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.0`
 - channels(user): `"darwin, home-manager, nixpkgs, nur-packages"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/Users/user/.nix-defexpr/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Closes #199220.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This should go to staging

**Comment by wegank** [comment]
... while leaving over 3800 packages broken on trunk?

**Comment by wegank** [comment]
@SuperSandro2000 Rebased to staging in #199338.

**Comment by vcunat** [comment]
I believe that master is more appropriate.  IMO failing packages don't really count into the rebuild amount.

File: pkgs/development/libraries/mesa/default.nix [hunk 1]
```diff
@@ -9,7 +9,7 @@
 , galliumDrivers ? ["auto"]
 , vulkanDrivers ? ["auto"]
 , eglPlatforms ? [ "x11" ] ++ lib.optionals stdenv.isLinux [ "wayland" ]
-, vulkanLayers ? [ "device-select" "overlay" ]
+, vulkanLayers ? lib.optionals stdenv.isLinux [ "device-select" "overlay" ]
```
Comment (Atemu) [inline pkgs/development/libraries/mesa/default.nix:12]: Could you explain what this does and why it fixes the darwin build (and ideally add that to the commit message)?

Is this really linux-specific or just a not-darwin thing?

**Review by Atemu — COMMENTED** [review]
Result of `nixpkgs-review pr 199223` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mesa</li>
  </ul>
</details>

File: pkgs/development/libraries/mesa/default.nix (hunk 1 above)
Comment (Atemu) [reply pkgs/development/libraries/mesa/default.nix:12]:
```suggestion
, vulkanLayers ? lib.optionals !stdenv.isDarwin [ "device-select" "overlay" ] # No Vulkan support on Darwin
```

Comment (wegank) [reply pkgs/development/libraries/mesa/default.nix]: Done.

**Review by Atemu — APPROVED** [review]

**Review by LunNova — APPROVED** [review]

**Review by marsam — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
> I believe that master is more appropriate. IMO failing packages don't really count into the rebuild amount.

This should have been written into the description. From my perspective the situation was not fully clear.

**Pull request merged**